widget/view.controller.js

```diff
@@ -9,37 +9,185 @@
     // Initialization variables
     $scope.defaultGlobalSettings = {};
-    $scope.updatedGlobalSettings = {};
+    $scope.defaultExclusionSettings = {};
+    $scope.updatedExclusionSettings = {};
+    $scope.defaultIOCTypeFieldMapping = {};
+    $scope.updatedIOCTypeFieldMapping = {};
     $scope.initList = [];
     $scope.searchString = '';
     $scope.searchStatus = 'off';
     $scope.isLightTheme = $rootScope.theme.id === 'light';
+    $scope.isSteelTheme = $rootScope.theme.id === 'steel';
     $scope.invalidIOCs = {}; // This dict holds invalid IOCs for various indicator types
+    $scope.fileName = '';
+    $scope.uploadedFileFlag = false;
+    $scope.loadingJob = false;
+    $scope.defaultIOCTypeList = [];
+    $scope.notYetEnteredIOCTypes = ['Add Custom Indicator Type'];
+    $scope.selectedIndicatorType = { iocType: '', pattern: [] };
+    $scope.addCustomIOCType = false;
+    $scope.isSystemIOCType = true;
+    $scope.iocTypeSelected = false;
+    const maxFileSize = 25072682;
+    var regexDict = {};
 
 
     // File Paths
     $scope.widgetCSS = widgetBasePath + 'widgetAssets/css/wizard-style.css';
-    $scope.startPageImage = $scope.isLightTheme ? widgetBasePath + 'images/sfsp-start-light.png' : widgetBasePath + 'images/sfsp-start-dark.png';
-    $scope.excludeIOCPageImage = widgetBasePath + 'images/sfsp-global-settings.png';
-    $scope.finishPageImage = widgetBasePath + 'images/finish.png';
+    $scope.pageImages = {
+      'startPageImage': $scope.isLightTheme ? widgetBasePath + 'images/ioc-extraction-start-light.png' : widgetBasePath + 'images/ioc-extraction-start-dark.png',
+      'excludeIOCPageImage': $scope.isLightTheme ? widgetBasePath + 'images/ioc-extraction-exclusion-light.png' : widgetBasePath + 'images/ioc-extraction-exclusion-dark.png',
+      'fieldMappingPageImage': $scope.isLightTheme ? widgetBasePath + 'images/ioc-extraction-field-map-light.png' : widgetBasePath + 'images/ioc-extraction-field-map-dark.png',
+      'finishPageImage': widgetBasePath + 'images/ioc-extraction-finish-both.png'
+    };
 
     // Functions
     $scope._buildPayload = _buildPayload;
     $scope.moveNext = moveNext;
     $scope.moveBack = moveBack;
-    $scope.commitGlobalSettings = commitGlobalSettings;
+    $scope.commitExclusionSettings = commitExclusionSettings;
     $scope.validateIOC = validateIOC;
     $scope.setSearchStatus = setSearchStatus;
     $scope.updateSearchQuery = updateSearchQuery;
     $scope._getRegexPattern = _getRegexPattern;
-    $scope._getKeyStoreValue = _getKeyStoreValue;
+    $scope.uploadFiles = uploadFiles;
+    $scope.setBulkImportFlags = setBulkImportFlags;
+    $scope.setAddNewIOCFlags = setAddNewIOCFlags;
+    $scope.getNotEnteredIOCTypes = getNotEnteredIOCTypes;
+    $scope.indicatorTypeChanged = indicatorTypeChanged;
+    $scope.saveNewIOCType = saveNewIOCType;
+
+
+    function saveNewIOCType() {
+      let keyStoreTemplate = iocExtractionConfigService.constants().keyStoreTemplate;
```

Comment:
You can create iocKeyStoreManager or iocKeyStoreHandler class to implement all keystore related operations, if not done already.

Reply:
Will explore this